system/modules/core/forms/FormFileUpload.php

```diff
@@ -238,6 +250,8 @@ public function validate()
 
 					// Generate the DB entries
 					$strFile = $strUploadFolder . '/' . $file['name'];
+
+					/** @var \FilesModel $objFile */
```

Comment:
No, it does not, because `findByPath` returns a collection (the path is not unique). Stumbled upon it as well.